src/guess_density.jl

```diff
@@ -1,32 +1,35 @@
 @doc raw"""
-    guess_density(basis)
+    guess_density(basis, magnetic_moments)
 
 Build a superposition of atomic densities (SAD) guess density.
 
 We take for the guess density a gaussian centered around the atom, of
 length specified by `atom_decay_length`, normalized to get the right number of electrons
 ```math
 \hat{ρ}(G) = Z \exp\left(-(2π \text{length} |G|)^2\right)
-```
+
+When magnetic moments are provided, construct a symmetry-broken density guess. The magnetic moments should be specified in units of ``μ_B``.
 """
-guess_density(basis::PlaneWaveBasis) = guess_density(basis, basis.model.atoms)
-@timing function guess_density(basis::PlaneWaveBasis{T}, atoms) where {T}
-    # Return sum of atomic densities with all weights equal to 1
-    gaussians = [(T(n_elec_valence(spec)), T(atom_decay_length(spec)), pos)
-                 for (spec, positions) in atoms for pos in positions]
-    gaussian_superposition(basis, gaussians)
+function guess_density(basis::PlaneWaveBasis, magnetic_moments=[])
+    guess_density(basis, basis.model.atoms, magnetic_moments)
 end
+@timing function guess_density(basis::PlaneWaveBasis{T}, atoms, magnetic_moments) where {T}
```

Comment:
I find this function a bit convoluted. I think what I would do is have a low-level internal function for the total density and one for the spin density and then `guess_density` just calls both and combines the result.

Reply:
good catch